sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -48,10 +49,11 @@ def get_hessians(y_true, raw_predictions):
     ('least_squares', -2., 42),
     ('least_squares', 117., 1.05),
     ('least_squares', 0., 0.),
-    # I don't understand why but y_true == 0 fails :/
-    # ('binary_crossentropy', 0.3, 0),
-    ('binary_crossentropy', -12, 1),
-    ('binary_crossentropy', 30, 1),
+    # The argmin of binary_crossentropy for y_true=±1 is ±inf due to logit,
+    # cf. "complete separation". Therefore we use 0 < y_true < 1.
```

Comment:
you mean 0 / 1 in stead of -1 / 1?

Also, why did it work with y_true == 1 then?

Reply:
Yes, I meant y = 0 gives -inf, y = 1 gives +inf.
The reason, it worked for `y_true = 1` so far was ... luck with numerical precision.

I looked at the solution of `newton`:
- For `y_true = 1` I find `optimum = 35.47761173`. Then `pointwise_loss` calls `np.logaddexp(0, optimum) - optimum` which gives `0` (true for all values of `optimum>33.271...`). 
- Then, for `y_true = 0`, the loss becomes `np.logaddexp(0, x)` which equals zero for `x<-745.1...`, which the Newton/Halley root finder does not find.